mix.exs

```diff
@@ -8,13 +8,15 @@ defmodule Phoenix.MixProject do
     end
   end
 
-  @version "1.6.4"
+  @version File.read!("VERSION")
```

Comment:
Doesn't this also require to use `@external_resource`

Reply:
it's an .exs file. It gets evaluated every time it is run. But I also wondered myself if there is a corner case where it should be required. Thank you for asking this.

Reply:
You're right it's an `.exs`. But I think something then isn't working as expected (at least with Elixir v1.13.4).
If it means `mix test` should also always reevaluate `mix.exs` then it's definitely not working for me.

I think probably some special behavior for `mix.exs` coming with Elixir v1.13 has introduces something that this does not work as expexted. Probably related to https://github.com/elixir-lang/elixir/issues/11704.

Reply:
It does not work with other elixir versions down to 1.10, either OTP 23 or 24,
Actually adding `@external_resource` does not solve the issue.

Reply:
Apparently this feature has never worked in Elixir
with external_resource: https://github.com/eksperimental/debug_me/runs/6055378901?check_suite_focus=true
without it: https://github.com/eksperimental/debug_me/actions/runs/2180695163

here's the script that checks for it.
https://github.com/eksperimental/debug_me/blob/db6e04feeab1f29c4dbddd5a100c40661c7e93fe/check.sh

Reply:
@cw789 The Mix tasks were storing the version number at compile time and that module didn't have a dependency on the VERSION file, so they file didn't get updated. Thanks @wojtekmach  for finding out the root cause.
Could you test it, I have, and it works as expected. Thank you.

Reply:
Perfect. I've learned some new insights. Thank you.